File: lib/shared/langsmith-config.ts
import { getConfig } from "../../bin/config";

export interface LangSmithConfig {
  enabled: boolean;
  apiKey?: string;
  project?: string;
  endpoint?: string;
}

export function loadLangSmithConfig(): LangSmithConfig {
  const config = getConfig();
  
  return {
    enabled: config.langsmith?.enabled || false,
    apiKey: config.langsmith?.apiKey,
    project: config.langsmith?.project || "genai-rag-workspace",
    endpoint: config.langsmith?.endpoint || "https://api.smith.langchain.com"
  };
}

File: lib/model-interfaces/langchain/functions/request-handler/langsmith_config.py
import os

def init_langsmith():
    """Initialize LangSmith tracing for LangChain"""
    enabled = os.getenv("LANGSMITH_ENABLED", "false").lower() == "true"
    if not enabled:
        return False

    # Set LangChain tracing environment variables
    os.environ["LANGCHAIN_TRACING_V2"] = "true"
    os.environ["LANGCHAIN_PROJECT"] = os.getenv("LANGSMITH_PROJECT", "genai-rag-workspace")
    
    api_key = os.getenv("LANGSMITH_API_KEY")
    if api_key:
        os.environ["LANGCHAIN_API_KEY"] = api_key
        
    os.environ["LANGCHAIN_ENDPOINT"] = os.getenv("LANGSMITH_ENDPOINT", "https://api.smith.langchain.com")
    
    print(f"[LangSmith] ✅ Initialized tracing for project: {os.getenv('LANGCHAIN_PROJECT')}")
    return True

# Initialize on import
langsmith_enabled = init_langsmith()
